src/rebar_pkg_resource.erl

```diff
@@ -124,6 +130,13 @@ download(TmpDir, Pkg={pkg, Name, Vsn, _Hash, Repo}, State, _ResourceState, Updat
 make_vsn(_, _) ->
     {error, "Replacing version of type pkg not supported."}.
 
+format_error({bad_registry_checksum, Name, Vsn, Expected, Found}) ->
+    io_lib:format("The checksum for package at ~ts-~ts (~ts) does not match the "
+                  "checksum expected from the registry (~ts). Either unlock and "
+                  "upgrade the package, or make sure you fetched it from "
+                  "the same repo from which the registry was fetched.",
```

Comment:
how would a user make sure to fetch it from the right repo?

Reply:
As far as I can figure the only way this could happen is if you changed the `repo_url` for a repo after having updated the registry for a specific package, resulting in the entry in the registry to have a checksum for the package that may differ from what is actually downloaded from the new repo url.

Maybe I should just remove that part? And instead change it to suggest running `rebar3 update <pkg>` since that would fetch new registry entries from the repo.

Reply:
yeah, "Run rebar3 update <pkg> and try again." would be a good actionable error message.

Reply:
k, went with, ```Run `rebar3 do unlock ~ts, update` and then try again.```